gmm.py

```diff
@@ -331,3 +433,60 @@ def __update_pi(self, pi):
         assert pi.size() in [(1, self.n_components, 1)], "Input pi does not have required tensor dimensions (%i, %i, %i)" % (1, self.n_components, 1)
 
         self.pi.data = pi
+
+    @staticmethod
+    def get_kmeans_mu(X, n_centers, min_delta=1e-3):
+        """
+        input:
+        x:              torch.Tensor (n, d),(n, 1, d)
+        output:
+        center:         torch.Tensor (1, k, d)
+        """
+        X = X.squeeze(1)
+        min, max = X.min(), X.max()
+        X = (X-min) / (max - min)
+        center = X[np.random.choice(np.arange(X.shape[0]), size=4, replace=False), ...]
+        delta = np.inf
+        while delta > min_delta:
+            l2_dis = torch.norm((X.unsqueeze(1).repeat(1,n_centers,1)-center),p=2,dim=2)
+            l2_cls = torch.argmin(l2_dis, dim=1)
+            center_old = center.clone()
+            for c in range(n_centers):
+                center[c] = X[l2_cls==c].mean(dim=0)
+            delta = torch.norm((center_old-center),dim=1).max()
+
+        return (center.unsqueeze(0)*(max-min)+min)
+
+if __name__=="__main__":
+    from math import sqrt
+    import matplotlib.pyplot as plt
+
+    n = 5000
+    n1 = 1000
+    K = 4
+
+    from sklearn.datasets import make_moons,make_blobs
```

Comment:
We should compare these results to `sklearn` (in particular for high dimensions).

A dedicated test would make sense – similar to `test.py:testEmMatchesSkLearn`. 😎